# Welcome to the BITMaB-2 workshop - Corpus Christi, TX (January 15-19, 2018)

This workshop provides training on how to describe the taxonomic diversity in a marine benthic community, using both morphological taxonomy and environmental sequencing approaches (genome assembly, metabarcoding amplicons, transcriptomics using RNA-seq, and other -Omics methods).

**Workshop Schedule** can be found [here](https://docs.google.com/document/d/1fztNYDuBBz93ZuKGAw-SGH066xMJ2lJx2_3fi6LDk6I/edit)

**Computer Preparation Instructions** for this workshop are available [online](https://docs.google.com/document/d/1VdYD8CFJvCIh1KDAz_9e09-Ipju8YJRfQ4RNdWkhI-U/edit#)

**Participant List** can be found [here](https://docs.google.com/spreadsheets/d/15G_CikjmffEvd0ASti5UzmHY9tem1iWmEbH3h1lDiAs/edit#gid=0)

Workshop Organizers:

* [W. Kelley Thomas](https://hcgs.unh.edu/) (PI), University of New Hampshire
* [Holly Bik](https://biklab.github.io) (Co-PI), University of California Riverside
* [Paul Montagna](https://www.harteresearchinstitute.org/people/paul-montagna) (Co-PI), Texas A&M University Corpus Christi
* Francesca Leasi, University of New Hampshire
* Krystalynne Morris, University of New Hampshire
* Michael Reuscher, University of New Hampshire

Invited Taxnomists: 

* Richard Heard, University of Southern Mississippi
* Alberto De Jésus Navarrete, ECOSUR-Chetumal
* Wonchoel Lee, Hanyang University
* Jesús Ángel de León González, Autonomous University of Nuevo León
* Fabio Moretzsohn, Texas A&M University Corpus Christi
* Tiago Pereira, University of California Riverside
* Alessandro Prantoni, Federal University of Paraná
* Adelaide Rhodes, University of Vermont
* Melissa Rohal, Texas A&M University Corpus Christi
* Martin Sørensen, Natural History Museum of Denmark

Bioinformatics team: 

* Taruna Schuelke, University of California Riverside
* Joseph Sevigny, University of New Hampshire
* Devin Thomas, University of New Hampshire

Fieldwork and Lab Technicians:

* Noe Barrera, Texas A&M University Corpus Christi
* Larry Hyde, Texas A&M University Corpus Christi
* Rick Kalke, Texas A&M University Corpus Christi
* Elani Morgan, Texas A&M University Corpus Christi
* Terry Palmer, Texas A&M University Corpus Christi
* Jamie Smith, Texas A&M University Corpus Christi

---
# Workshop Lectures and Tutorials

**Monday January 15, 2018**

* [Introduction to Genomics, Bioinformatics, and Sequencing Technologies](https://github.com/BikLab/BITMaB2-Tutorials/tree/master/presentations/KelleyThomas-Introduction-15Jan2018.pdf) (Kelley Thomas)
* Introduction to the command line environment and the world of BASH (Devin Thomas and Joe Sevigny)
	* [BASH intro commands](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/bash_lesson.sh) - first part of printout
	* [QIIME2 common commands](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/bash_lesson.sh) - second part of printout
	* BASH command cheatsheet - third part of printout
* Advanced command line tricks
* **Taxonomy Lecture**: [Goals of the workshop and how it fits into the context of the research project “Genomic Responses to the Deepwater Horizon event and development of high-throughput biological assays for oil spills”](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/FrancescaLeasi-BiologicalAssays-15Jan2018.pdf) (Dr. Francesca Leasi)
* **Taxonomy Lecture**: [Bivalvia and Gastropoda of the GoM](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/FabioMoretzsohn-Bivalves-and-Gastropods-15Jan2018.pdf) (Dr. Fabio Moretzsohn)

**Tuesday January 16, 2018**

* [Genome assembly and annotation](https://github.com/BikLab/BITMaB2-Tutorials/tree/master/presentations/KelleyThomas-Introduction-15Jan2018.pdf) (Kelley Thomas)
* **Taxonomy Lecture**: [Nematoda - ecology and species diversity](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/AlbertodeJes%C3%BAsNavarrete-Nematoda-16Jan2018.pdf) (Dr. Alberto de Jesús Navarrete)
* **Taxonomy Lecture**: [Nematoda - molecular phylogeny and metabarcoding](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/TiagoPereira-Nematoda-16Jan2018.pdf) (Dr. Tiago Pereira)

**Wednesday January 17, 2018**

* Introduction to the QIIME2 pipeline for microbial ecology and metabarcoding data analysis (Dr. Holly Bik)
	* [QIIME2 Overview Slides](http://bit.ly/q2-vegas-2017) (from June 2017 QIIME2 workshop)
	* [QIIME1 to QIIME2 script conversion cheatsheet](https://docs.google.com/spreadsheets/d/1AOyA84lGPBk5C2Uflg_pILEL75a6_gdY6SK3ltOt44E/edit#gid=0)
	* [QIIME 2 workflow for metabarcoding analysis (18S/16S rRNA) with already-demultiplexed fastq files](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/QIIME2-metabarcoding-tutorial-already-demultiplexed-fastqs.md)
* **Taxonomy Lecture**: [Harpacticoida](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/WonchoelLee-Copepoda-17Jan2018.pdf) (Dr. Wonchoel Lee)
* **Taxonomy Lecture**: [Kinorhyncha & Gnathostomulida](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/MartinSorensen-Gnathostomulida-and-Kinorhyncha-15Jan2018.pdf) (Dr. Martin Sørensen)

**Thursday January 18, 2018**

* Data analysis and visualization in QIIME2, continued (Dr. Holly Bik)
* Discussion and Q&A session about genome assembly results and data analysis
* **Taxonomy Lecture**: [Polychaeta](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/MichaelReuscher-Polychaete-18Jan2018.pdf) (Dr. Jesús Ángel de León González or Dr. Michael Reuscher)
* **Taxonomy Lecture**: [Oligochaeta](https://github.com/BikLab/BITMaB2-Tutorials/blob/master/presentations/AlessandroPrantoni-Oligochaeta-18Jan2018.pdf) (Dr. Alessandro Prantoni)

**Friday January 19, 2018**

* Mapping DNA sequence reads to reference genomes to investigate genetic diversity or profiling gene expression (Dr. Kelley Thomas, Devin Thomas, and Joe Sevigny)
* Genome visualizations using the UCSC Genome Browser (Dr. Kelley Thomas, Devin Thomas, and Joe Sevigny)
* Introduction to Whole Metagenome Shotgun Analysis -  principles, data sets, methods, and introduction to PALADIN (Drs. Kelley Thomas and Holly Bik)
* **Taxonomy Lecture**: [Cumacea and Tanaidacea, with emphasis on shallow water]() (Dr. Richard Heard)
* **Taxonomy Lecture**: [Cumacea and Tanaidacea, with emphasis on deep-sea]() (Dr. Adelaide Rhodes)

---

# Some General Notes and Suggestions on Computational Training:

The [Software Carpentry](https://software-carpentry.org/) and Data Carpentry organizations are a wonderful resource for learning general programming skills. They run in person bioinformatics workshps at institutes worldwide, and maintain open source lesson plans of all workshop materials. These lessons are continually updated as bioinformatics and software tools evolve (which happens frequently!)

We suggest starting with these keystone Carpentry lessons:

* [The Unix Shell](http://swcarpentry.github.io/shell-novice/)
* [Version Control with Git](http://swcarpentry.github.io/git-novice/)
* [R for Reproducible Scientific Analysis - Software Carpentry](http://swcarpentry.github.io/r-novice-gapminder/)
* [Programming with R](http://swcarpentry.github.io/r-novice-inflammation/)
* [Programming with Python - Software Carpentry](http://swcarpentry.github.io/python-novice-inflammation/)
* [Data Carpentry Ecology Curriculum](http://www.datacarpentry.org/lessons/#ecology-workshop)
* [Data Carpentry Genomics Curriculum](http://www.datacarpentry.org/lessons/#genomics-workshop)

Also, try to find a good way to take notes (bioinformatics commands and comments about what they are doing and what parameters you should think about). We reccomend doing this using simple text files (e.g. with Markdown notation) instead of proprieary software such as Microsoft word that does not play nice with copy/paste because of hidden characters.

Some reccommendations for note taking tools:

* [Mou](http://25.io/mou/) (markdown text editor)
* [Sublime](https://www.sublimetext.com/) (markdown / programming text editor)
* [TextWrangler](https://www.barebones.com/products/textwrangler/) (basic text editor with programming features)
* [Jupyter Notebooks](http://jupyter.org/) (interactive browswer-based programming notebooks with built in visualzation capabilities)


